File: base/src/main/java/com/yxr/base/util/DisplayUtil.java
package com.yxr.base.util;

import com.yxr.base.BaseApplication;

public class DisplayUtil {
    public static int getScreenWidth() {
        return BaseApplication.context.getResources().getDisplayMetrics().widthPixels;
    }

    public static int getScreenHeight() {
        return BaseApplication.context.getResources().getDisplayMetrics().heightPixels;
    }

    /**
     * convert px to its equivalent dp
     * <p>
     * 将px转换为与之相等的dp
     */
    public static int px2dp(float pxValue) {
        final float scale = BaseApplication.context.getResources().getDisplayMetrics().density;
        return (int) (pxValue / scale + 0.5f);
    }


    /**
     * convert dp to its equivalent px
     * <p>
     * 将dp转换为与之相等的px
     */
    public static int dp2px(float dipValue) {
        final float scale = BaseApplication.context.getResources().getDisplayMetrics().density;
        return (int) (dipValue * scale + 0.5f);
    }


    /**
     * convert px to its equivalent sp
     * <p>
     * 将px转换为sp
     */
    public static int px2sp(float pxValue) {
        final float fontScale = BaseApplication.context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (pxValue / fontScale + 0.5f);
    }


    /**
     * convert sp to its equivalent px
     * <p>
     * 将sp转换为px
     */
    public static int sp2px(float spValue) {
        final float fontScale = BaseApplication.context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (spValue * fontScale + 0.5f);
    }
}

File: base/src/main/java/com/yxr/base/util/DisplayUtil.kt
package com.yxr.base.util

import com.yxr.base.BaseApplication

class DisplayUtil {
    companion object{
        @JvmStatic
        fun getScreenWidth(): Int {
            return BaseApplication.context.resources.displayMetrics.widthPixels
        }

        @JvmStatic
        fun getScreenHeight(): Int {
            return BaseApplication.context.resources.displayMetrics.heightPixels
        }

        /**
         * convert px to its equivalent dp
         * 将px转换为与之相等的dp
         */
        @JvmStatic
        fun px2dp(pxValue: Float): Int {
            val scale = BaseApplication.context.resources.displayMetrics.density
            return (pxValue / scale + 0.5f).toInt()
        }

        /**
         * convert dp to its equivalent px
         * 将dp转换为与之相等的px
         */
        @JvmStatic
        fun dp2px(dipValue: Float): Int {
            val scale = BaseApplication.context.resources.displayMetrics.density
            return (dipValue * scale + 0.5f).toInt()
        }

        /**
         * convert px to its equivalent sp
         * 将px转换为sp
         */
        @JvmStatic
        fun px2sp(pxValue: Float): Int {
            val fontScale = BaseApplication.context.resources.displayMetrics.scaledDensity
            return (pxValue / fontScale + 0.5f).toInt()
        }

        /**
         * convert sp to its equivalent px
         * 将sp转换为px
         */
        @JvmStatic
        fun sp2px(spValue: Float): Int {
            val fontScale = BaseApplication.context.resources.displayMetrics.scaledDensity
            return (spValue * fontScale + 0.5f).toInt()
        }
    }
}